src/main/java/org/opensearch/security/configuration/MaskedField.java

```diff
@@ -57,31 +60,31 @@ public final void isValid() throws Exception {
     }
 
     public byte[] mask(byte[] value) {
-        if (isDefault()) {
-            return blake2bHash(value);
+        if (algo != null) {
+            return customHash(value, algo);
+        } else if (regexReplacements != null) {
+            String cur = new String(value, StandardCharsets.UTF_8);
+            for (RegexReplacement rr : regexReplacements) {
+                cur = cur.replaceAll(rr.getRegex(), rr.getReplacement());
+            }
+            return cur.getBytes(StandardCharsets.UTF_8);
+        } else if (StringUtils.isNotEmpty(defaultAlgorithm)) {
```

Comment:
Does the order of execution matter here? If so why choose this particular order?

Reply:
Yes the order is important. The logic is the same as the existing code (albeit slightly refactored) with the addition of the new configuration option. 

Previously the masking choice as detailed in https://opensearch.org/docs/latest/security/access-control/field-masking/#advanced-use-an-alternative-hash-algorithm is basically to: 
a) use what is configured for the field(s) according to the role definition if explicitly specified 
OR 
b) default to Blake2b where nothing explicitly set

Now it is 
a) use what is configured for the field(s) according to the role definition if explicitly specified 
OR 
b) use the new configuration to specify an algorithm other than Blake2b to default to without having to do this via every role definition
OR
c) default to existing, backwards compatible Blake2b where nothing explicitly set either in the role definition or configuration